### DISCUSSION TITLE
增加划词翻译和翻译面板

### DISCUSSION THREAD

**Discussion by luzov** [immersive-translate/immersive-translate#1077] [Ideas]
### 新功能 | New Feature

1. 划词翻译
2. 翻译面板

### 为什么您想要添加这些功能 | Why you want to add these features

1. 有些只是部分单词不认识需要查询
2. 翻译面板是希望可以支持我自己输入一些内容进行翻译

**Comment by theowenyoung** [comment]
目前沉浸式翻译想尽可能保持简单一点，所以建议使用其他划词翻译的软件做补充。

**Comment by luzov** [comment]
😂我就是有点强迫症不想多装几个插件才提的。有什么推荐吗？

**Reply by theowenyoung** [reply]
沙拉查词

**Comment by douglarek** [comment]
期待实现此功能，和沉浸式翻译类似的插件 Tancy 支持划词，但不支持自定义 openai

**Reply by luzov** [reply]
刚看，现在好像支持自定义openai了，不过我还希望能像沉浸式翻译一样支持其他翻译api

**Reply by douglarek** [reply]
嗯，这个功能是这个插件非常基础的能做到的功能，可能是觉得这个功能和沉浸式无关

**Comment by KnIfER** [comment]
强烈建议将翻译功能暴露给用户脚本（翻译某个dom元素），这样用户可以输入内容。然后调用翻译。

It is highly recommended to expose the translation functionality to the user script (translating a DOM element) so that the user can enter content. Then call the translation

好吧现在就有（以上），只是只能翻译输入框内容（快捷键），而且把原文删了？hack一下可以实现“划词翻译和翻译面板 ”
